tfe/resource_tfe_policy_set.go

```diff
@@ -0,0 +1,300 @@
+				Optional: true,
+			},
+
+			"global": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"organization": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"policies": &schema.Schema{
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"workspaces": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceTFEPolicySetCreate(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+	global := d.Get("global").(bool)
+	organization := d.Get("organization").(string)
+	policies := d.Get("policies").(*schema.Set)
+	workspaces := d.Get("workspaces").(*schema.Set)
+
+	// First off, make sure they aren't trying to supply workspaces for a global set
+	if global && workspaces.Len() > 0 {
+		err := errors.New(`cannot set "workspaces" when "global" is true`)
+		return fmt.Errorf(
+			"Error creating policy set %s for organization %s: %v", name, organization, err)
+	}
+
+	// Create a new options struct.
+	options := tfe.PolicySetCreateOptions{
+		Name:        tfe.String(name),
+		Description: tfe.String(description),
+		Global:      tfe.Bool(global),
+	}
+
+	// Set up the policies and workspaces.
+	for _, policyID := range policies.List() {
+		options.Policies = append(options.Policies, &tfe.Policy{ID: policyID.(string)})
+	}
+
+	if !*options.Global {
+		for _, workspaceID := range workspaces.List() {
+			options.Workspaces = append(options.Workspaces, &tfe.Workspace{ID: workspaceID.(string)})
+		}
+	}
+
+	log.Printf("[DEBUG] Create policy set %s for organization: %s", name, organization)
+	policySet, err := tfeClient.PolicySets.Create(ctx, organization, options)
+	if err != nil {
+		return fmt.Errorf(
+			"Error creating policy set %s for organization %s: %v", name, organization, err)
+	}
+
+	d.SetId(policySet.ID)
+
+	return resourceTFEPolicySetRead(d, meta)
+}
+
+func resourceTFEPolicySetRead(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	log.Printf("[DEBUG] Read policy set: %s", d.Id())
+	policySet, err := tfeClient.PolicySets.Read(ctx, d.Id())
+	if err != nil {
+		if err == tfe.ErrResourceNotFound {
+			log.Printf("[DEBUG] Policy set %s does no longer exist", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading policy set %s: %v", d.Id(), err)
+	}
+
+	// Update the config.
+	d.Set("name", policySet.Name)
+	d.Set("description", policySet.Description)
+	d.Set("global", policySet.Global)
+
+	// Update the relationships.
+	var policies []interface{}
+	for _, policy := range policySet.Policies {
+		policies = append(policies, policy.ID)
+	}
+	d.Set("policies", policies)
+
+	if !policySet.Global {
```

Comment:
Do we need to check this? I.e. does `policySet.Workspaces` contain any workspaces if its a global policy set? I would expect it to be empty and so the `range` op would be a no-op.

Reply:
The API will actually always return workspaces for a policy set. This is because global policy sets are still technically related to workspaces (in this case, all of the organization's workspaces) even if not having been _directly_ associated to workspaces by the user.

So this is still necessary, as we don't want to directly set this attribute in the provider for global policy sets